README.md

```diff
@@ -552,19 +586,44 @@ var config = {
 * [tokenManager](#tokenmanager)
   * [tokenManager.add](#tokenmanageraddkey-token)
   * [tokenManager.get](#tokenmanagergetkey)
+  * [tokenManager.getTokens](#tokenmanagergettokens)
+  * [tokenManager.setTokens](#tokenmanagersettokenstokens)
   * [tokenManager.remove](#tokenmanagerremovekey)
   * [tokenManager.clear](#tokenmanagerclear)
   * [tokenManager.renew](#tokenmanagerrenewkey)
   * [tokenManager.on](#tokenmanageronevent-callback-context)
   * [tokenManager.off](#tokenmanageroffevent-callback)
+* [authStateManager](#authstatemanager)
+  * [authStateManager.getAuthState](#authstatemanagergetauthstate)
+  * [authStateManager.updateAuthState](#authstatemanagerupdateauthstate)
+  * [authStateManager.subscribe](#authstatemanagersubscribehandler)
+  * [authStateManager.unsubscribe](#authstatemanagerunsubscribehandler)
 
 ------
 
 ### `signIn(options)`
 
 > :hourglass: async
 
-The goal of an authentication flow is to [set an Okta session cookie on the user's browser](https://developer.okta.com/use_cases/authentication/session_cookie#retrieving-a-session-cookie-by-visiting-a-session-redirect-link) or [retrieve an `id_token` or `access_token`](https://developer.okta.com/use_cases/authentication/session_cookie#retrieving-a-session-cookie-via-openid-connect-authorization-endpoint). The flow is started using `signIn`.
+This method supports both `browser-based OpenID Connect` and `custom signIn with session cookie` flows by accepting different signInOptions. If the provided options are not valid for `custom signIn with session cookie`, this method fallbacks to the `browser-based OpenID Connect` flow.
+
+#### browser-based OpenID Connect flows
+
+This flow Calls `onAuthRequired` function if it was set on the initial configuration. Otherwise, it will call [loginRedirect](#loginRedirectfromuri-additionalparams). In this flow, there is a fromUri parameter in options to push the user to after successful authentication, and the addtional params are mapped to the [Authorize options](#authorize-options).
+
+For more information on the options, see the [loginRedirect](#loginRedirectfromuri-additionalparams) method below.
+
+```javascript
+// Start the browser based oidc flow, then parse tokens from the redirect callback url
+authClient.signIn();
+
```

Comment:
There is a redirect that happens here, right? I think these two lines of code would be in two different parts of the app, separated by some state logic. Either the app has loaded to the point where it is calling `signIn` or it is loading to the point of calling `handleAuthentication()` based upon some other information. Could we add a link to the `isLoginRedirect` method here.

Logically, its like:
```
if (authClient.token.isLoginRedirect()) {
  authClient.handleAuthentication();
} else if (!isAuthenticated()) {
authClient.signIn();
} else {
// authenticated experience
}
```